src/exercises.js

```diff
@@ -1,36 +1,50 @@
-function filterNumbers(arr, largerThan) {
+/*function filterNumbers(arr, largerThan) {
   return arr.filter(function (number) {
     return number > largerThan;
   });
+}*/
+
+const filterNumbers = (arr,largerThan)=> {
+  return arr.filter((number)=>{
+    return number > largerThan
+  })
 }
 
-function returnFirstTwoDrivers() {
+
+/*function returnFirstTwoDrivlargerThaners() {
+  const drivers = ["Sally", "Bob", "Freddy", "Claudia"];
+  return drivers.slice(0, 2);
+}*/
+
+const returnFirstTwoDrivers = ()=> {
   const drivers = ["Sally", "Bob", "Freddy", "Claudia"];
   return drivers.slice(0, 2);
 }
 
 const person = {
   name: "Bob",
-  greet: function () {
+  greet: () =>{
     setTimeout(
       function () {
-        console.log("Hello, my name is " + this.name);
-        console.log(this);
-      }.bind(this),
-      2000
+        console.log("Hello, my name is " + person.name);
+        console.log(person);
+      }.bind(person),2000
     );
-  },
+ return "Hello, my name is"+ this.name
+ },
 };
 
-function add(a, b) {
-  a = a || 1;
-  b = b || 1;
+function add(a = 1, b = 1) {
   return a + b;
 }
 
 function createGrid(array) {
-  return `Generates a ${array[0]} x ${array[1]} grid`;
+  const [x,y, ...resto] = array
+  /*return `Generates a (${[x]}) x (${[y]}) grid`;*/
+  return "Generates a " + array[x] + " x" + array[y] + " grid"
+ 
```

Comment:
function createGrid([width, height]) {
  return `Generates a ${width} x ${height} grid`;
}

function sum(...args) {
  return args.reduce((a, b) => a + b, 0);
}

function addNumbers(...numbers) {
  return numbers.reduce((sum, number) => sum + number, 0);
}

function calculate(product, ...numbers) {
  return numbers.map((number) => number * product);
}

function join(array1, array2) {
  return [...array1, ...array2];
}

const savedFile = {
  extension: "jpg",
  name: "repost",
  size: 14040,
};

function fileSummary({ name, extension, size }) {
  return `The file ${name}.${extension} is of size ${size}`;
}

function addressMaker(address) {
  const { city, state, country } = address;

  const newAddress = {
    city,
    state,
    country,
  };

  return `${newAddress.city}, ${newAddress.state}, ${newAddress.country}`;
}

function Ninja(options) {
  this.name = options.name;
  this.health = 100;
  this.speed = 3;
  this.strength = 3;
}

Ninja.prototype.sayName = function () {
  console.log("My ninja name is " + this.name);
};

Ninja.prototype.showStats = function () {
  console.log(
    "Name: " +
      this.name +
      ", Health: " +
      this.health +
      ", Speed: " +
      this.speed +
      ", Strength: " +
      this.strength
  );
};

Ninja.prototype.drinkSake = function () {
  this.health += 10;
  console.log("Health: " + this.health);
};